bidict/_common.py

```diff
@@ -1,14 +1,61 @@
 """
 Implements :class:`BidirectionalMapping`, the bidirectional map base class.
 
-Also provides related exception classes.
+Also provides related exception classes and collision behaviors.
 """
 
-from .compat import PY2, iteritems
-from .util import pairs
+from .compat import PY2, iteritems, viewkeys
+from .util import inverted, pairs
 from collections import Mapping
 
 
+def _proxied(methodname, ivarname='_fwd', doc=None):
+    """Make a func that calls methodname on the indicated instance variable."""
+    def proxy(self, *args):
+        ivar = getattr(self, ivarname)
+        meth = getattr(ivar, methodname)
+        return meth(*args)
+    proxy.__name__ = methodname
+    proxy.__doc__ = doc or 'Like :py:meth:`dict.%s`.' % methodname
+    return proxy
+
+
+class CollisionBehavior(object):
+    """
+    Provide RAISE, OVERWRITE, and IGNORE collision behaviors.
+
+    .. py:attribute:: RAISE
+
+        Raise an exception when a collision is encountered.
+
+    .. py:attribute:: OVERWRITE
+
+        Overwrite an existing item when a collision is encountered.
+
+    .. py:attribute:: IGNORE
+
+        Keep the existing item and ignore the new item when a collision is
+        encountered.
+
+    """
+
+    def __repr__(self):
+        return '<CollisionBehavior:%s>' % self.__class__.__name__
+
+class RAISE(CollisionBehavior):
+    """Raise an exception when a collision is encountered."""
+
+class OVERWRITE(CollisionBehavior):
+    """Overwrite an existing item when a collision is encountered."""
+
+class IGNORE(CollisionBehavior):
+    """Ignore the new item when a collision is encountered."""
+
+CollisionBehavior.RAISE = RAISE = RAISE()
```

Comment:
I looked into that and it didn't seem worth it. IIRC, the reasoning was it would add a dependency, would not make the code any more concise, and I think the documentation generated wouldn't be quite as nice. Meanwhile, it wouldn't add anything over what this is already doing.

What am I missing about the benefits of doing this?